interop-tests/src/arch.rs

```diff
@@ -47,87 +35,97 @@ pub(crate) mod native {
-                format!("/ip4/{ip}/udp/0/quic-v1"),
-            ),
-            (Transport::Tcp, Ok(SecProtocol::Tls)) => (
-                tcp::tokio::Transport::new(tcp::Config::new())
-                    .upgrade(Version::V1Lazy)
-                    .authenticate(tls::Config::new(&local_key).context("failed to initialise tls")?)
-                    .multiplex(muxer_protocol_from_env()?)
-                    .timeout(Duration::from_secs(5))
-                    .boxed(),
-                format!("/ip4/{ip}/tcp/0"),
-            ),
-            (Transport::Tcp, Ok(SecProtocol::Noise)) => (
-                tcp::tokio::Transport::new(tcp::Config::new())
-                    .upgrade(Version::V1Lazy)
-                    .authenticate(
-                        noise::Config::new(&local_key).context("failed to intialise noise")?,
-                    )
-                    .multiplex(muxer_protocol_from_env()?)
-                    .timeout(Duration::from_secs(5))
-                    .boxed(),
-                format!("/ip4/{ip}/tcp/0"),
-            ),
-            (Transport::Ws, Ok(SecProtocol::Tls)) => (
-                WsConfig::new(tcp::tokio::Transport::new(tcp::Config::new()))
-                    .upgrade(Version::V1Lazy)
-                    .authenticate(tls::Config::new(&local_key).context("failed to initialise tls")?)
-                    .multiplex(muxer_protocol_from_env()?)
-                    .timeout(Duration::from_secs(5))
-                    .boxed(),
-                format!("/ip4/{ip}/tcp/0/ws"),
-            ),
-            (Transport::Ws, Ok(SecProtocol::Noise)) => (
-                WsConfig::new(tcp::tokio::Transport::new(tcp::Config::new()))
-                    .upgrade(Version::V1Lazy)
-                    .authenticate(
-                        noise::Config::new(&local_key).context("failed to intialise noise")?,
-                    )
-                    .multiplex(muxer_protocol_from_env()?)
-                    .timeout(Duration::from_secs(5))
-                    .boxed(),
-                format!("/ip4/{ip}/tcp/0/ws"),
-            ),
-            (Transport::WebRtcDirect, _) => (
-                webrtc::tokio::Transport::new(
-                    local_key,
-                    webrtc::tokio::Certificate::generate(&mut rand::thread_rng())?,
-                )
-                .map(|(peer_id, conn), _| (peer_id, StreamMuxerBox::new(conn)))
-                .boxed(),
-                format!("/ip4/{ip}/udp/0/webrtc-direct"),
-            ),
+        behaviour_constructor: impl FnOnce(&Keypair) -> B,
+    ) -> Result<(Swarm<B>, String)> {
+        let (swarm, addr) = match (transport, from_env::<SecProtocol>("security")) {
+            (Transport::QuicV1, _) => {
+                let swarm = libp2p::SwarmBuilder::with_new_identity()
+                    .with_tokio()
+                    .with_quic()
+                    .with_behaviour(behaviour_constructor)?
+                    .build();
```

Comment:
Note that this is using a 10s instead of the previous 5s timeout now.

I argue that 10s is good enough for almost all cases. Thus I only want to make it configurable in case there is demand.